### ISSUE TITLE
Changing matplotlib parameter inside Jupyter Notebooks

### ISSUE THREAD

**Issue by a-monsch** [dsavoiu/kafe2#167]
**Describe the bug**
kafe2 seems to modify within Jupyter Notebboks some of the rcParams already during the import. It shows different behavior for - in this case an `ipywidget` - certain matplotlib applications.

**To Reproduce**
Package versions used:
* `numpy==1.20.1`
* `ipywidgets==7.6.3`
* `matplotlib==3.3.4`
* `kafe2==2.2.3`
* `IPython==7.21.0`

Important for the reproduction: For the execution an `jupyter notebook` server must be started and not an `jupyter lab` server (because the latter often have some problems with ìpywidgets`).

Code to be run inside a Notebook cell:
```
import kafe2 as k2  # Comment this out for the different result

import matplotlib.pyplot as plt
import ipywidgets as ipw
import numpy as np

def ui():
    slider_1 = ipw.FloatSlider(description="a", value=1, min=-1, max=1)
    slider_2 = ipw.FloatSlider(description="b", valueue=0, min=-1, max=1)
    _out = ipw.interactive_output(plot, {"slider_1": slider_1, "slider_2": slider_2})
    return ipw.VBox([ipw.HBox([slider_1, slider_2]), _out])

def plot(**kwargs):
    fig, ax = plt.subplots(1, 1, figsize=(10, 10))
    _x = np.linspace(-1, 1, 100)
    ax.plot(_x, float(kwargs["slider_1"]) * _x + float(kwargs["slider_2"]))
    ax.hlines(0, -1, 1, color="black")
    ax.vlines(0, -1, 1, color="black")
    ax.set_xlim(-1, 1)
    ax.set_ylim(-1, 1)
    fig.canvas.draw()

ui()
```

**Expected behavior**
kafe2 should set any `matplotlib` parameters only in case of plotting (and afterwards - if it is not already the situation - switch back to the previous settings afterwards)

**System:**
 - OS: Windows10

 - Python Version: 3.8.1

**Comment by cverstege** [comment]
Thanks for bringing this to our attention. This might be similar/have the same cause as #9 .
I will take a closer look at the matplotlib issues in about one week.
Hopefully this can be fixed by only using rc_context, whenever using matplotlib internally. I will investigate this further and update this issue once I'm working on it.

I hope I can finish this work and incorporate a fix for the upcoming 2.3. release.

**Comment by cverstege** [comment]
@a-monsch I can't reproduce the issue locally. The plot always looks the same for me. What exactly are the differences for you?

Im running the exaxt same versions as you, but Python 3.9 on Linux.

**Comment by cverstege** [comment]
I think i might found the problem, I always use
`%matplotlib inline` before importing anything. This forces kafe2 to use the internal ipython backend instead of the default one.
Can you please retest on your end with `%matplotlib inline` at the beginning of the cell?
Maybe we should default to the ipython backend (when running in jupyter/ipython) and fallback to the TkAgg backend if it's not available.

**Comment by cverstege** [comment]
@a-monsch can you please re-test with the current dev branch? I've pushed some changes, to the backend choosing in 48f697fd7370a56bb44198e310442bb0a68dad5e

Now there should be no need to use `%matplotlib inline` anymore as kafe2 will now use the iPython backend if available.

**Comment by cverstege** [comment]
I've forced push a new and simple fix in c96a53c3d646f38b19cd88cc0bc28a925322efad. Please test with the latest and greatest version from the dev branch.

**Comment by a-monsch** [comment]
> I've forced push a new and simple fix in [c96a53c](https://github.com/dsavoiu/kafe2/commit/c96a53c3d646f38b19cd88cc0bc28a925322efad). Please test with the latest and greatest version from the dev branch.

I have tested the example I gave above and can confirm that the previous error no longer occurs.


Thanks for the fix!

**Comment by cverstege** [comment]
Thanks. The fix will be included in the upcoming 2.3 release.
